### ISSUE TITLE
Peer Review - Windows

### ISSUE THREAD

**Issue by TiagoF99** [OpenLiberty/draft-guide-operator-openshift#7]
## Peer Review conducted on 3 platforms: Mac, Windows & Linux
- Mac OS is our base OS as the dev content team generally develops on the Mac OS.
- Conduct three separate peer reviews for each OS. 

Peer Review: review to be done by a peer member. 

Functionality
---
- [x] Check that the guide has working instructions and sample code
- [x] If any URL visits, try them on different browsers: Firefox, Chrome, Safari
- [x] If any URL visits, try `curl` command where applicable 

Formatting & Presentation
---

`README.adoc` checks:
- [x] Ensure that command substitutions use the `$(cmd)` format and not ``` `cmd` ```
- [ ] Ensure that the console output of the integrated tests do not include the `[INFO]` tab
- [x] Check the quality and presentation of guide according to [Structure and Style Guideline](https://github.com/OpenLiberty/draft-guides-template/wiki/Guidelines-for-Structure-and-Styles)
- [x] Check that the pom.xml, server.xml, etc files are clean
    - [x] Remove any dependencies and features that are not required
    - [x] Ensure that the parent pom.xml files do not contain any dependencies, and all dependencies are specified in the children pom.xml files
- [x] Examples with the right outcomes are provided
- [x] Example with the wrong outcomes, if any, are provided

`pom.xml` checks (if files are present):
- [x] Check that the file follows the template found here: https://github.com/OpenLiberty/draft-guides-template/blob/master/finish/pom.xml
- [x] Check that the following dependencies are included in all of the `pom.xml` files:
```
<!-- Provided dependencies -->
<dependency>
  <groupId>jakarta.platform</groupId>
  <artifactId>jakarta.jakartaee-api</artifactId>
  <version>8.0.0</version>
  <scope>provided</scope>
</dependency>

<dependency>
  <groupId>org.eclipse.microprofile</groupId
  <artifactId>microprofile</artifactId>
  <version>3.3</version>
  <type>pom</type>
  <scope>provided</scope>
</dependency>
```
- [x]  Check that the versions of plugins are directly provided and not fed in by a file property.
  - Not formatted like so:
```
    <properties>
        <!-- Plugins -->
        <version.liberty-maven-plugin>3.2</version.liberty-maven-plugin>
    </properties>

    <!-- Liberty plugin -->
    <plugin>
        <groupId>io.openliberty.tools</groupId>
        <artifactId>liberty-maven-plugin</artifactId>
        <version>${version.liberty-maven-plugin}</version>
    </plugin>
```
  - Should be formatted like so:
```
    <!-- Liberty plugin -->
    <plugin>
        <groupId>io.openliberty.tools</groupId>
        <artifactId>liberty-maven-plugin</artifactId>
        <version>3.2</version>
    </plugin>
```
- [x] Check that there are 4 spaces per indent for proper formatting
- [x] Ensure that the guide is using the latest version of the `liberty-maven-plugin` or `liberty-gradle-plugin` where applicable

Overall checks:
- [x] Check the consistency of guide with the template and other guides
- [x] Check the quality of code according to the best coding practices
- [x] Check that all licensing statements are properly stated in all files, with the correct year (Should be present in all Java files + the `index.html`)
- [x] Check that the directories are properly structured
- [x] Check that some of these `page-tags` are used in a guide: `MicroProfile, Maven, Docker, Kubernetes, Gradle, Java EE, Security, Cloud`. Only these tags are visible on the website. Latest list [here](https://github.com/OpenLiberty/openliberty.io/blob/master/src/main/content/_includes/visible-tags.liquid). 
- [x] Check the `attribution` statement is accurate for the guide
- [x] Verify the `travisTest.sh` script, if any, is accurate and consistent with other guides

---

Guide’s contributor’s (if available, otherwise peer tester’s) responsibility: 
- [ ] Additional tests where applicable: 
    - [ ] Define test coverage and review with team (including guide contributor, if available)
    - [ ] Define detail test cases 
    - [ ] Consider corner cases targeting the specific guide
    - [x] Consider corner cases UI tests
    - [x] Consider testing URL on all browsers, ie, FF, Chrome, Safari
    - [x] Consider testing the `curl` command for URL visits
- [ ] Consider building with both Maven and Gradle build tools 
- [x] Testing with different IDEs, ie, Atom, Eclipse (Optional: VS.code, IntelliJ, Microclimate)
- [ ] Run Acrolinx Checker on draft (above 70 score approximately)
- [x] Consider SEO title and description for the guides 
- [x] Ensure automated test is enable, set up with Travis CI, and able to schedule tests 
- [x] Run `diff -r start/ finish/` and there's no differences
- [x] Ensure that the automation tests are able to run when PR is created

Peer Tester’s responsibility: 
- [x] Check the appearance of the guide on test site for the following items: 
    - [x] Table of contents 
    - [x] Headings
    - [x] Paragraphs
    - [x] code snippets
    - [x] outputs
    - [x] links
    - [x] hotspots 
- [x] Test the guide end-to-end with working instruction and sample code
- [x] Perform all the defined test cases

**Comment by TiagoF99** [comment]
Under Deploying the Kafka service to OpenShift:
"Apache Kafka is the messaging broker used in this application. The image required to deploy the application is available and can be pulled and deployed directly from DockerHub.

To deploy Zookeeper, run the following command:"
- Consider explaining what Zookeeper is for better clarity to what the commands following it are actually doing

**Comment by TiagoF99** [comment]
"Run the following commands to view the build logs:

oc logs build/system-buildconfig-1
oc logs build/inventory-buildconfig-1

Once you see the Push successful message, run the following command to view the image streams:"

- my output:

[root@tira-inf start]# oc logs build/system-buildconfig-1
Receiving source from STDIN as archive ...
Caching blobs under "/var/cache/blobs".

Pulling image openliberty/open-liberty:kernel-java8-openj9-ubi ...
Getting image source signatures
Copying blob sha256:78afc5364ad2c981e4a4919f535aaefef9ac2f990837be01c766764e025b1f31
Copying blob sha256:a1bcbf077941c52e57cec56ffd68f10f14c7f511beac5e2ca750238302ea2ab1
Copying blob sha256:58e1deb9693dfb1704ccce2f1cf0e4d663ac77098a7a0f699708a71549cbd924
Copying blob sha256:287ff2820835b8bab88bf7e1a2cea999bc197efa843e0e6fc850fb482af95420
Copying blob sha256:4016e18dee776dd6ba9a2a874c1f28e500f333acf6ef7b8980b1f78e19c4331a
Copying blob sha256:158e24b1291b43d876cd314f8b5861b6c7f732ac37aa9f91c3fd1c4b9079d59b
Copying blob sha256:b96f01bd60fcc84dbdca1b48f63ce8bb0a7999df4b3c7111073c9c04102de4ff
Copying blob sha256:aa3d79791201351463c8ad41e0d5bdaca082af756afd4e3bfe2e38e8319c16ac
Copying blob sha256:27365c94ffa5c283a21af36fd6bcb026df050c91d63293edb6caa9d87ff5e12e
Copying blob sha256:8cb4fca82f64ad2c02723c2ffc9db63c728f822400024b8999d0185d33a50226
Copying blob sha256:395529862db170d821ac614d7c2b44ff8e6d238de495207452e2a94ee456f746
Copying config sha256:5c7bcd0a758e0a07141b7130d7f5932c098d72d2885af35553241265eeb5133f
Writing manifest to image destination
Storing signatures
[root@tira-inf start]# oc logs build/inventory-buildconfig-1
Receiving source from STDIN as archive ...
Caching blobs under "/var/cache/blobs".

Pulling image openliberty/open-liberty:kernel-java8-openj9-ubi ...
Getting image source signatures
Copying blob sha256:a1bcbf077941c52e57cec56ffd68f10f14c7f511beac5e2ca750238302ea2ab1
Copying blob sha256:4016e18dee776dd6ba9a2a874c1f28e500f333acf6ef7b8980b1f78e19c4331a
Copying blob sha256:78afc5364ad2c981e4a4919f535aaefef9ac2f990837be01c766764e025b1f31
Copying blob sha256:58e1deb9693dfb1704ccce2f1cf0e4d663ac77098a7a0f699708a71549cbd924
Copying blob sha256:158e24b1291b43d876cd314f8b5861b6c7f732ac37aa9f91c3fd1c4b9079d59b
Copying blob sha256:287ff2820835b8bab88bf7e1a2cea999bc197efa843e0e6fc850fb482af95420
Copying blob sha256:27365c94ffa5c283a21af36fd6bcb026df050c91d63293edb6caa9d87ff5e12e
Copying blob sha256:aa3d79791201351463c8ad41e0d5bdaca082af756afd4e3bfe2e38e8319c16ac
Copying blob sha256:b96f01bd60fcc84dbdca1b48f63ce8bb0a7999df4b3c7111073c9c04102de4ff
Copying blob sha256:8cb4fca82f64ad2c02723c2ffc9db63c728f822400024b8999d0185d33a50226
Copying blob sha256:395529862db170d821ac614d7c2b44ff8e6d238de495207452e2a94ee456f746
Copying config sha256:5c7bcd0a758e0a07141b7130d7f5932c098d72d2885af35553241265eeb5133f
Writing manifest to image destination
Storing signatures
STEP 1: FROM openliberty/open-liberty:kernel-java8-openj9-ubi
STEP 2: ARG VERSION=1.0
6b2888e15221f1af9c5178d0e7a61cf3d846d7c1395423d7de556a313e098070
STEP 3: ARG REVISION=SNAPSHOT
6646afc88332b3067765085f7d93ade071548425313713e8817aa96a34995164
STEP 4: LABEL org.opencontainers.image.authors="Your Name" org.opencontainers.image.vendor="Open Liberty" org.opencontainers.image.url="local" org.opencontainers.image.source="https://github.com/OpenLiberty/guide-operator-openshift" org.opencontainers.image.version="$VERSION" org.opencontainers.image.revision="$REVISION" vendor="Open Liberty" name="inventory" version="$VERSION-$REVISION" summary="The inventory microservice from the 'Deploying microservices using the Open Liberty Operator on OpenShift' guide" description="This image contains the inventory microservice running with the Open Liberty runtime."
760c42791ab9cb45178d5e2e64a6ea075a7f544244a9309c904fb6fbf0a19e48
STEP 5: COPY --chown=1001:0 src/main/liberty/config /config/

- No "Push Succesful" Message after following guide yet following outlined command works

**Comment by TiagoF99** [comment]
"
Create the deploy.yaml configuration file.
deploy.yaml
"

- Consider adding "Create the deploy.yaml configuration file in start directory"

**Comment by TiagoF99** [comment]
under scaling up the system application: 
"Run the following command to update the system resource:

oc apply -f deploy.yaml

You see the following output:
openlibertyapplication.openliberty.io/system created
openlibertyapplication.openliberty.io/inventory unchanged"

- When I ran it on windows I got:
openlibertyapplication.openliberty.io/system configured
openlibertyapplication.openliberty.io/inventory unchanged

**Comment by TiagoF99** [comment]
Under accessing the inventory application, when the command "oc get routes" is run the output is "No resources found in guide namespace." and not the expected output outlined in the guide. All the other commands not highlighted in a previous comment work and output as expected.

**Comment by TiagoF99** [comment]
- consider removing white space inbetween dependencies in the pom.xml files as well as white space in the index.htnl file inbetween the /</p/> and /<div/>

**Comment by TiagoF99** [comment]
- In inventoryManager.java consider combining lines 39 and 40 for shorter code as such:
    if (A && B) {} instead of if (A) {if (B) {}}

**Comment by TiagoF99** [comment]
- You have these tags (// tag::assert2[] for example) in InventoryServiceIT.java but not in any other files. Makes it seem very inconsistent.

**Comment by TiagoF99** [comment]
[] when verifying these two, a PR was created but no travis tests were run:
    - Ensure automated test is enable, set up with Travis CI, and able to schedule tests
    - Ensure that the automation tests are able to run when PR is created

**Comment by MaiHameed** [comment]
> No "Push Successful" Message after following guide yet following outlined command works

The build takes a while to finish, so the message would only show up once you check the logs after it's finished. I made it a little more clear in the README with the following sentence:

> Wait until the build completes. You will know it is complete once you see the `Push successful` message. Run the following command to view the image streams:

> Consider adding "Create the deploy.yaml configuration file in start directory"

The relative path to the file is displayed inside the command block under the instructions. in this case it's `deploy.yaml` which means that it should be created in the current directory, aka `start`

> Under accessing the inventory application, when the command "oc get routes" is run the output is "No resources found in guide namespace." and not the expected output outlined in the guide. All the other commands not highlighted in a previous comment work and output as expected.

You need to give it a minute for the deployments to be fully live

> consider removing white space inbetween dependencies and plugins in the pom.xml files as well as white space in the index.html file inbetween the <\p> and <div>

Minor style comment, I'll leave as is.

> Ensure automated test is enable, set up with Travis CI, and able to schedule tests

This is being addressed with this PR #9

**Comment by TiagoF99** [comment]
[] - I think that "Wait until the build completes. You will know it is complete once you see the Push successful message" should come before " Run the following commands to view the build logs: oc logs build/system-buildconfig-1 oc logs build/inventory-buildconfig-1" and it should be updated. Like it should be something like:



The local system and inventory directories will be uploaded to OpenShift to be built into the Docker images. Run the following command to list the builds and track their status:

-----------------------------
oc get builds

You see the following output:

NAME                      TYPE     FROM             STATUS     STARTED
system-buildconfig-1      Docker   Binary@f24cb58   Running    45 seconds ago
inventory-buildconfig-1   Docker   Binary@f24cb58   Running    13 seconds ago

You may need to wait some time until the build is complete. To check for this, run the following commands to view the build logs until a "push succesful" message appears:

oc logs build/system-buildconfig-1
oc logs build/inventory-buildconfig-1

Run the following command to view the image streams:

oc get imagestreams

....

**Comment by TiagoF99** [comment]
- [ ] oc get routes still gives no output after running through the guide again and waiting a few minutes. This seems to be an issue with other OS' as well. May need to be investigated.

**Comment by MaiHameed** [comment]
Addressed with #10
